### PR TITLE
Nguyen Ngoc Thanh - mocks

### PR DESCRIPTION
**Pull request by nthanhster**

### CODE REVIEW

File: src/test/java/cz/inventi/kpj/kpjtesting/KpjTestingTest.java [hunk 1]
```diff
@@ -10,41 +19,40 @@
...
 
+  @BeforeEach
   void setUp() {
-    // setup mocks for all tests
-    // hint you have to use doReturn for Spy
     // Change behavior of HelloWorldServiceImpl#echo(String)
-    // to always returns "KPJ Rulez"
-
+    // to always return "KPJ Rulez"
+    doReturn("KPJ Rulez").when(helloWorldService.echo("Print ECHO"));
```
Comment (Saljack) [inline src/test/java/cz/inventi/kpj/kpjtesting/KpjTestingTest.java]: It is wrong syntax. You can use calling method inside a method only for `Mockito.when()`.
`Mockito.verify()` and `Mockito.doXXX.when()` these methods return T and you call the method on T. 

For example you want to mock `foo.bar()` then you call `verify(foo).bar()`

```suggestion
    doReturn("KPJ Rulez").when(helloWorldService).echo("Print ECHO");
```

**Comment by Saljack** [comment]
Tests do not work. You use `verify` and `doXXX.when` wrong

**Review by nthanhster — COMMENTED** [review]
I have incorporated the comments

**Comment by nthanhster** [comment]
Ahoj Tomasi, nevis, proc mi failuji checks, i prestoze jsem implementoval zmeny?

File: src/test/java/cz/inventi/kpj/kpjtesting/KpjTestingTest.java [hunk 2]
```diff
@@ -10,41 +19,40 @@
  * 4. Change behavior of {@link HelloWorldServiceImpl#echo(String)} to always
  * returns "KPJ Rulez" <br/>
  */
+@SpringBootTest
```
Comment (Saljack) [inline src/test/java/cz/inventi/kpj/kpjtesting/KpjTestingTest.java]: Remove it. This test is without Spring. Try to use just Mockito. These tests are faster and you can test isolated code.
```suggestion
```

**Pull request closed without merging**